src/main/java/com/iexec/resultproxy/ipfs/IpfsService.java

```
@@ -40,33 +66,43 @@ public Optional<byte[]> get(String ipfsHash) {
         return Optional.empty();
     }
 
-    public String add(String fileName, byte[] fileContent) {
-        NamedStreamable.ByteArrayWrapper file = new NamedStreamable.ByteArrayWrapper(fileName, fileContent);
+    public String add(final String fileName, final byte[] fileContent) {
+        final NamedStreamable.ByteArrayWrapper file = new NamedStreamable.ByteArrayWrapper(fileName, fileContent);
         try {
-            MerkleNode pushedContent = ipfs.add(file, false).get(0);
+            final MerkleNode pushedContent = ipfs.add(file, false).get(0);
             return pushedContent.hash.toString();
         } catch (IOException e) {
             log.error("Error when trying to push ipfs object [fileName:{}]", fileName);
         }
         return "";
     }
 
-    public static boolean isIpfsHash(String hash) {
+    public static boolean isIpfsHash(final String hash) {
         try {
             return Multihash.fromBase58(hash).toBase58() != null;
         } catch (Exception e) {
             return false;
         }
     }
 
+    private String convertHostToIp(final String hostname) {
+        InetAddress address = null;
+        try {
+            address = InetAddress.getByName(hostname);
+        } catch (UnknownHostException e) {
+            log.error("No IP address could be found [host:{}]", hostname, e);
+        }
+        return address != null ? address.getHostAddress() : "";
```

I don't think we want this method to return an empty string. It will never create a valid multi-address as defined by `this.multiAddress = "/ip4/" + ipfsNodeIp + "/tcp/" + port`
In order to fail fast, I would prefer to throw an exception from this method.

---

Resolved in [eed8f2d](https://github.com/iExecBlockchainComputing/iexec-result-proxy/pull/155/commits/eed8f2dd6ba49099966d5c093bc994b331e707c3)